packages/server/src/scim.ts

```diff
@@ -0,0 +1,126 @@
+import { DefaultScimProvider, ScimConfig, ScimUserRequestData } from "@padloc/core/src/scim";
+import { MemberProvisioning, Provisioner } from "@padloc/core/src/provisioning";
+
+import { createServer, IncomingMessage, ServerResponse } from "http";
+import { readBody } from "./transport/http";
+
+export class ScimProvider extends DefaultScimProvider {
+    constructor(public readonly config: ScimConfig, public readonly provisioner: Provisioner) {
+        super(config, provisioner);
+    }
+
+    async init() {
+        await this._startScimServer();
+    }
+
+    private _getDataFromScimRequest(httpReq: IncomingMessage) {
+        const url = new URL(`http://localhost${httpReq.url || ""}`);
+        const secretToken = url.searchParams.get("token") || "";
+        const orgId = url.searchParams.get("org") || "";
+
+        // TODO: find account/org based on token
```

Comment:
@MaKleSoft any idea of where should this (token) be? I'm thinking of automatically generating + adding it to the org when it's created, but how would it be searchable here?

Reply:
You can find the `OrgProvisioning` and `Org` objects based on `orgId`:

```sh
const orgProv = await this._getOrCreateOrgProvisioning(orgId);
const org = await this.storage.get(Org, orgId);
```

And the `AccountProvisioning`:

```sh
const accountProv = this._getOrCreateAccountProvisioning({ email });
```

To use `_getOrCreateOrgProvisioning` and `_getOrCreateAccountProvisioning` you'll have to subclass `BasicProvisioner` and make those methods `protected` instead of `private`

Reply:
You can find the `OrgProvisioning` and `Org` objects based on `orgId`:

```sh
const orgProv = await this._getOrCreateOrgProvisioning(orgId);
const org = await this.storage.get(Org, orgId);
```

And the `AccountProvisioning`:

```sh
const accountProv = this._getOrCreateAccountProvisioning({ email });
```

To use `_getOrCreateOrgProvisioning` and `_getOrCreateAccountProvisioning` you'll have to subclass `BasicProvisioner` and make those methods `protected` instead of `private`

Reply:
So the secret token wouldn't be used for the lookup but just as an additional check. You can store it on the `OrgProvisioning` object for now.

```ts
if (orgProv.scimSecret !== secret) {
     throw new Err(ErrorCode.AUTHENTICATION_FAILED, "...")
}
```